crates/ty_python_semantic/src/types/display.rs

```diff
@@ -847,18 +848,33 @@ impl<'db> FmtDetailed<'db> for DisplayRepresentation<'db> {
             Type::Intersection(intersection) => intersection
                 .display_with(self.db, self.settings.clone())
                 .fmt_detailed(f),
-            Type::IntLiteral(n) => write!(f, "{n}"),
-            Type::BooleanLiteral(boolean) => f.write_str(if boolean { "True" } else { "False" }),
+            Type::IntLiteral(n) => write!(f.with_detail(TypeDetail::Type(self.ty)), "{n}"),
+            Type::BooleanLiteral(boolean) => f
+                .with_detail(TypeDetail::Type(self.ty))
+                .write_str(if boolean { "True" } else { "False" }),
             Type::StringLiteral(string) => {
-                write!(f, "{}", string.display_with(self.db, self.settings.clone()))
+                write!(
+                    f.with_detail(TypeDetail::Type(self.ty)),
+                    "{}",
+                    string.display_with(self.db, self.settings.clone())
+                )
             }
+            // an alternative would be to use `Type::SpecialForm(SpecialFormType::LiteralString)` here,
+            // which would mean users would be able to jump to the definition of `LiteralString` from the
+            // inlay hint, but that seems less useful than the definition of `str` for a variable that is
+            // inferred as an *inhabitant* of `LiteralString` (since that variable will just be a string
+            // at runtime)
             Type::LiteralString => f
                 .with_detail(TypeDetail::Type(self.ty))
                 .write_str("LiteralString"),
```

Comment:
Hmm I think I disagree? From what I've seen LiteralString only shows up in code that's explicitly declared e.g. a function argument as LiteralString, but maybe I've just been lucky? (not a strong opinion)